docker/kayobe/Dockerfile

```diff
@@ -7,6 +7,11 @@ ARG BASE_IMAGE="quay.io/centos/centos:stream8"
 FROM ${BASE_IMAGE}
 MAINTAINER "Will Szumski" <will@stackhpc.com>
 
+ARG http_proxy
+ARG https_proxy
```

Comment:
We already pass these args in when building the image in [SKC CI](https://github.com/stackhpc/stackhpc-kayobe-config/blob/402ce66c4b3820c3949243d70f45d2aa4af21def/.github/workflows/stackhpc-build-kayobe-image.yml#L81). Presumably they were unused.

Reply:
I also see that http(s)_proxy are referenced on L69, perhaps coming from Docker config?

Reply:
If that is the case then we would need to move them before download packages as SMSLab kayobe-automation cannot build unless these proxy settings are present.

Reply:
Let's wait for Will to get the history on this.

Reply:
CC @jovial

Reply:
Sure. I think we also need to consider that the container is running without `network=host` which would have impact on this and is the reason for changes when compared to SKC CI?

Reply:
I think I did once bake in the the proxy environment variables as env vars, but I recalled I had issues when using the same image in another environment (which used another proxy). Passing these variables in at runtime seemed like the better option. Is this the only way to set a proxy at build time? I thought there were some command line parameters to docker build :thinking:

Reply:
I will investigate further. Good point about sharing the image between environment and still carrying the proxy settings.

Reply:
If we could make them not persistent after build that would be ideal. Can we get away with the build arg only and possibly pass them in like this:

```
RUN http_proxy=$http_proxy https_proxy=$http_proxy ./script
```

or possibly you can clear the env var?

Reply:
Missed the comment in https://github.com/stackhpc/kayobe-automation/pull/45/files#r1440221535, so that code is still in there. I guess this isn't so bad then.

Reply:
I think this is the magic that means that you don't need to explicitly define the build arg: https://docs.docker.com/engine/reference/builder/#predefined-args